packages/blocks/src/api/registration.js

```diff
@@ -25,12 +25,313 @@ import { isValidIcon, normalizeIconObject } from './utils';
 import { DEPRECATED_ENTRY_KEYS } from './constants';
 
 /**
- * Render behavior of a block type icon; one of a Dashicon slug, an element,
- * or a component.
+ * The `edit` function describes the structure of your block in the context of
+ * the editor. This represents what the editor will render when the block is
+ * used.
  *
- * @typedef {(string|WPElement|WPComponent)} WPBlockTypeIconRender
+ * @callback FnEditBlock
  *
- * @see https://developer.wordpress.org/resource/dashicons/
+ * @param {WPBlockEditProps} props
+ * @return {WPComponent}
+ */
+
+/**
+ * The `save` function defines the way in which the different attributes should
+ * be combined into the final markup, which is then serialized into
+ * `post_content`.
+ *
+ * @callback FnSaveBlock
+ *
+ * @param {WPBlockSaveProps} props
+ * @return {WPComponent}
+ */
+
+/**
+ * @callback FnSetAttributes
+ *
+ * @param {WPBlockAttributes} attributes Attributes to set.
+ * @return {void}
```

Comment:
FWIW, many conventions support omitting `@return` when the return value is `undefined` (including TypeScript?), and I think this is the general approach we've taken in the rest of the project.

>If a function that is not in externs has no return value, you can omit the `@return` tag, and the compiler will assume that the function returns `undefined`.

https://developers.google.com/closure/compiler/docs/js-for-compiler#tag-return

Reply:
Just chiming in here for a bit of added distinction...

As far as typescript is concerned, a function that returns `undefined` must return either `undefined` or simply `return;` for all code paths.

A function that returns `void` may or may not return `undefined` somewhere, but does not need to `return` in all code paths.